cpp/core/utils/compression.h

```diff
@@ -23,8 +23,13 @@ namespace gluten {
 
 enum CodecBackend { NONE, QAT, IAA };
 
+// ONE_BY_ONE mode will preallocate max compressed buffer, and then compress each buffer to the max compressed buffer
+// Stream mode will coly the buffers to a big buffer and then compress the big buffer
+enum CompressionMode { ONE_BY_ONE, STREAM };
+
 std::unique_ptr<arrow::util::Codec> createArrowIpcCodec(
     arrow::Compression::type compressedType,
     CodecBackend codecBackend);
 
+CompressionMode getCompressionMode(CodecBackend codecBackend);
```

Comment:
We should also pass compression codec here. QAT supports both gzip and zstd codec. We only need to copy buffers for gzip codec.